packages/ra-ui-materialui/src/detail/TabbedShowLayout.js

```diff
@@ -87,17 +87,19 @@ const TabbedShowLayout = props => {
         ...rest
     } = props;
     const match = useRouteMatch();
-
     const classes = useStyles(props);
+    const finalChildren = Array.isArray(children)
+        ? children.filter(child => child !== null)
+        : children;
 
     return (
         <div className={className} key={version} {...sanitizeRestProps(rest)}>
-            {cloneElement(tabs, {}, children)}
+            {cloneElement(tabs, {}, finalChildren)}
 
             <Divider />
             <div className={classes.content}>
-                {Children.map(children, (tab, index) =>
-                    tab && isValidElement(tab) ? (
+                {Children.map(finalChildren, (tab, index) => {
```

Comment:
here, you can use the regular `children`, right? This avoids having a different value at each render, and defeating pure component optimizations.

Reply:
I don't understand, if I go back to using `Children.map(children, (tab, index) =>` the tests won't pass.
Should I filter `nonNullChildren` with `useCallback`?